pay-api/src/pay_api/services/eft_service.py

```diff
@@ -121,14 +123,40 @@ def create_shortname_refund(cls, request: Dict[str, str], **kwargs) -> Dict[str,
         current_app.logger.debug(f'Starting shortname refund : {shortname_id}')
 
         refund = cls._create_refund_model(request, shortname_id, amount, comment)
-        recipients = EFTRefundEmailList.find_all_emails()
+        cls._refund_eft_credits(shortname_id, amount)
 
+        recipients = EFTRefundEmailList.find_all_emails()
         subject = f'Pending Refund Request for Short Name {shortname}'
         html_body = cls._render_email_body(shortname, amount, comment, shortname_id)
 
         send_email(recipients, subject, html_body, **kwargs)
         refund.save()
 
+    @classmethod
+    def _refund_eft_credits(cls, shortname_id: str, amount: str):
+        """Refund the amount to eft_credits table based on short_name_id."""
+        refund_amount = Decimal(amount)
+        eft_credits = EFTCreditModel.find_by_short_name_id(int(shortname_id))
+
+        for credit in eft_credits:
+            if refund_amount <= 0:
+                break
+            credit_amount = Decimal(credit.remaining_amount)
+            if credit_amount <= 0:
+                continue
+
+            if refund_amount <= credit_amount:
+                credit.remaining_amount -= refund_amount
+                refund_amount = Decimal(0)
+            else:
+                refund_amount -= credit_amount
+                credit.remaining_amount = Decimal(0)
+
+            credit.save()
+
+        if refund_amount > 0:
+            current_app.logger.error(f'Shortname {shortname_id} Refund amount exceed eft_credits remaining amount.')
```

Comment:
At this point you may have already done a save() even though there is not enough credits. We should probably roll back if conditions aren't met? My suggestion for the validation above should prevent this.

Should add a test for the business exception